### ISSUE TITLE
Type hinting for tuples

### ISSUE THREAD

**Issue by CEXT-Dan** [microsoft/pyright#8001]
Hi,  I’m having trouble getting type hinting for tuples to work properly. The issue is I’m making wrappers using boost and sometimes the most I can do is have the wrapper stub out a generic tuple. 

This had worked at one time, so I’m not sure if this is a regression 
Pylance v2024.5.1

Any hints?

```Python
import traceback
from pyrx_imp import Rx, Ge, Gi, Db, Ap, Ed

# boost::python::tuple makeSelectionResult(const ads_name& name, Acad::PromptStatus result)
# {
#     PyAutoLockGIL lock;
#     return boost::python::make_tuple<Acad::PromptStatus, PyEdSelectionSet>(result, PyEdSelectionSet{ name });
# }

# @staticmethod
# def select (filer: list='None')-> tuple :
#     ...

def foo()->tuple:
    return ("hello world", 1)

def PyRxCmd_doit() -> None:
    try:
        db = Db.curDb()
        filter = [(Db.DxfCode.kDxfStart, "LINE")]
        ssres: tuple[Ed.PromptStatus,Ed.SelectionSet] = Ed.Editor.select(filter)
        if ssres[0] != Ed.PromptStatus.eOk:
            raise RuntimeError("Selection Error! {}: ".format(ssres[0])) 
        
        ssres[0].#nothing
        
    except Exception as err:
        traceback.print_exception(err)

def PyRxCmd_doit2() -> None:
    try:
        val : tuple[str,int] = foo()
        val[0].#nothing
    except Exception as err:
        traceback.print_exception(err)

```

**Comment by CEXT-Dan** [comment]
Oh, thanks for that!
As a workaround, I can make my stub generator output` def foo()->tuple[Any, ...]:`
And that works!

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
